src/pip/_internal/commands/freeze.py

```diff
@@ -83,12 +84,18 @@ def run(self, options, args):
 
         cmdoptions.check_list_path_option(options)
 
+        paths = options.path
+        # Filter sys.path, to avoid listing distributions from
+        # current directory
+        if paths is None:
+            paths = [item for item in sys.path if item and item != os.getcwd()]
```

Comment:
Does this actually work? `sys.path[0]` is `''` (empty string) for me, while `os.getcwd()` returns an absolute path. This would probably be

```python
paths = [item for item in sys.path if item and item not in (os.curdir, '')]
```

Reply:
A `''` entry would not be considered as, through the first part of `if` statement i.e. `if ''` would be False.

Reply:
Oh sorry I missed the `if item` part. I do not think `sys.path` contains an absolute version of cwd by default though (it does not for me).

Reply:
It does not actually. However, when you'll print `sys.path` within the `freeze` operation or in `get_installed_distribution()` method, it somehow does.

On the contrary, when I print `sys.path` within the `WorkingSet` defined in the `pkg_resource`  it does not.

Reply:
That’s weird. We’ll probably need to figure out why it’s the case before we can come up with a proper fix.

Reply:
Hi there, how about
```python
paths = [item for item in sys.path if os.path.abspath(item) != os.path.abspath(os.getcwd())]
```
since neither of [sys.path](https://docs.python.org/3/library/sys.html#sys.path) nor [os.getcwd()](https://docs.python.org/3/library/os.html#os.getcwd) is warrantied to always give absolute paths.

Reply:
@McSinyx, the problem is that somehow `sys.path` contains absolute path of CWD apart from `''`, which is added by default when you use `python -m`. We have to show the package during freeze if the full path is present in `sys.path` but not when just `''` is present.